x/mongo/driver/errors.go

```diff
@@ -331,16 +331,16 @@ func (e Error) NamespaceNotFound() bool {
 	return e.Code == 26 || e.Message == "ns not found"
 }
 
-// helper method to extract an error from a reader if there is one; first returned item is the
-// error if it exists, the second holds parsing errors
-func extractError(rdr bsoncore.Document) error {
+// ExtractErrorFromBsoncoreDoc extracts an error from a bsoncore.Document if there is one.
+// Also used in testing for SDAM.
+func ExtractErrorFromBsoncoreDoc(doc bsoncore.Document) error {
```

Comment:
We want functions names to describe their purpose in as much detail as possible. This function name is technically correct, but it doesn't provide details about what the bsoncore document should actually contain. Also, the function API already takes `bsoncore.Document`, so `BsoncoreDoc` in the function name doesn't provide any new information. Instead, the name should describe the document's semantics and invariants. I'd suggest renaming to something like `ExtractErrorFromServerResponse`. Thoughts?

Reply:
`ExtractErrorFromServerResponse` seems appropriate to me; all three current usages seem to do exactly that, and it's true that `BsoncoreDoc` is redundant.